File: app/src/main/java/com/kieran/winnipegbusbackend/UpcomingStop.java
package com.kieran.winnipegbusbackend;

import android.support.annotation.NonNull;

public class UpcomingStop extends Stop implements Comparable {
    private StopTime time;
    private ScheduledStopKey key;

    public UpcomingStop(String stopName, int stopNumber, StopTime time, ScheduledStopKey key) {
        super(stopName, stopNumber);
        this.time = time;
        this.key = key;
    }

    public UpcomingStop(Stop stop, StopTime time, ScheduledStopKey key) {
        super(stop);
        this.time = time;
        this.key = key;
    }

    public StopTime getTime() {
        return time;
    }

    public ScheduledStopKey getKey() {
        return key;
    }

    @Override
    public int compareTo(@NonNull Object another) {
        return key.getStopNumber() - ((UpcomingStop)another).getKey().getStopNumber();
    }

}

File: app/src/main/java/com/kieran/winnipegbusbackend/UpcomingStop.kt
package com.kieran.winnipegbusbackend

import com.kieran.winnipegbusbackend.winnipegtransit.WinnipegTransitScheduledStopKey

class UpcomingStop : Stop, Comparable<Any> {
    var time: StopTime? = null
        private set
    var key: WinnipegTransitScheduledStopKey? = null
        private set

    constructor(stopName: String, stopNumber: Int, time: StopTime, key: WinnipegTransitScheduledStopKey) : super(stopName, stopNumber) {
        this.time = time
        this.key = key
    }

    constructor(stop: Stop, time: StopTime, key: WinnipegTransitScheduledStopKey) : super(stop) {
        this.time = time
        this.key = key
    }

    override operator fun compareTo(other: Any): Int {
        if(key == null|| other !is UpcomingStop || other.key == null)
            return Int.MIN_VALUE

        return key!!.stopNumber - other.key!!.stopNumber
    }

}
